csrc/flash_attn/src/flash_fwd_launch_template.h

```diff
@@ -142,7 +144,9 @@ void run_mha_fwd_hdim32(Flash_fwd_params &params, cudaStream_t stream) {
     constexpr static int Headdim = 32;
     BOOL_SWITCH(params.p_dropout < 1.f, Is_dropout, [&] {
         BOOL_SWITCH(params.is_causal, Is_causal, [&] {
-            run_flash_fwd<Flash_fwd_kernel_traits<Headdim, 128, 128, 4, false, false, T>, Is_dropout, Is_causal>(params, stream);
+            BOOL_SWITCH(params.is_alibi, Is_alibi, [&] {
```

Comment:
If you're already templating in "run_flash_fwd", you don't need to template here (and the rest of the file).